src/der.rs

```diff
@@ -17,11 +17,19 @@
 //! This module contains the foundational parts of an ASN.1 DER parser.
 
 use untrusted;
-use error;
 
 pub const CONSTRUCTED: u8 = 1 << 5;
 pub const CONTEXT_SPECIFIC: u8 = 2 << 6;
 
+/// An error in ASN1 parsing. Normal hygiene applies: if parsing secret
+/// DER, do not differentiate this from any other error.
+#[derive(Clone, Copy)]
+pub struct ASN1Error;
+
+impl From<untrusted::EndOfInput> for ASN1Error {
+    fn from(_: untrusted::EndOfInput) -> Self { ASN1Error }
+}
```

Comment:
I don't think this is correct. It is true that *in this module*, any time we encounter end-of-input, it's because of bad ASN.1. But, in generally it doesn't make sense to convert *any* or *all* `EndOfInput` errors to `ASN1Error`. Instead it has to be decided on a case-by-case basis.

Instead, we probably need to create wrapper functions around `Reader::read_byte()` and friends that map `EndOfInput` to `ASN1Error`, and use them here instead of using `Input`'s methods directly.